anjali/sb-pt-cont/9.js

```diff
@@ -0,0 +1,231 @@
+    numSpritesClicked: 0,
+    numSpriteClicksCauseSpeech: 0
+  };
+}
+
+// Helper variables
+var spriteIds = getSpriteIdsInUse();
+var eventLog = getEventLog();
+
+if (!validationProps.previous) {
+  validationProps.previous = {
+    waitTime: 210,
+    numStartingSprites: spriteIds.length
+  };
+}
+
+// Check for at least 3 different sprites
+validationProps.successCriteria.threeSprites = spriteIds.length>=3;
+
+// When program starts
+if (World.frameCount == 1) {
+  // Check that no sprites are touching
+  for (var i=0; i<spriteIds.length; i++) {
+    for (var j=i+1; j<spriteIds.length; j++) {
+      if (isTouchingSprite({id: spriteIds[i]}, {id: spriteIds[j]})) {
+        validationProps.successCriteria.noSpritesTouching = false;
+        break;
+      }
+    }
+    if (!validationProps.successCriteria.noSpritesTouching) {
+      break;
+    }
+  }
+  
+  // Check sprite locations
+  var uniqueStartingSpriteLocations = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var coords = [getProp({id: spriteIds[i]}, "x"), getProp({id: spriteIds[i]}, "y")];
+    var noDuplicateCoords = true;
+    for (var j=0; j<uniqueStartingSpriteLocations.length; j++) {
+      if ((coords[0] == uniqueStartingSpriteLocations[j][0]) &&
+          (coords[1] == uniqueStartingSpriteLocations[j][1])) {
+        noDuplicateCoords = false;
+        break;
+      }
+    }
+    if (!noDuplicateCoords) {
+      break;
+    } else {
+      uniqueStartingSpriteLocations.push(coords);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteLocations.length) {
+    validationProps.successCriteria.differentLocations = true;
+  }
+
+  // Check sprite costumes
+  var uniqueStartingSpriteCostumes = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var costume = getProp({id: spriteIds[i]}, "costume");
+    var noDuplicateCostumes = true;
+    for (var j=0; j<uniqueStartingSpriteCostumes.length; j++) {
+      if (costume == uniqueStartingSpriteCostumes[j]) {
+        noDuplicateCostumes = false;
+        break;
+      }
+    }
+    if (!noDuplicateCostumes) {
+      break;
+    } else {
+      uniqueStartingSpriteCostumes.push(costume);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteCostumes.length) {
+    validationProps.successCriteria.differentCostumes = true;
+  }
+}
+
+// Check if student is interacting with the program
+// version 1
+for (var spriteId in spriteIds) {
+  if (getProp({id: spriteId}, "timeout")>0) {
+    validationProps.waitTime = World.frameCount + 60;
+    break;
+  } else {
+    validationProps.waitTime = validationProps.previous.waitTime;
+  }
+}
+
+/*
+// version 2
+if (validationProps.successTime && (eventLog.length > validationProps.previous.eventLogLength)) {
+  validationProps.waitTime = World.frameCount + 100;
+} else {
+  validationProps.waitTime = validationProps.previous.waitTime;
+}
+*/
+
+// Keeps track of which sprites were clicked on
+if(!validationProps.spritesClickedOn) {
+  validationProps.spritesClickedOn = [];
```

Comment:
Should this be nested under `previous`? Like `validationProps.previous.spritesClickedOn`

Reply:
add all lists/sets that change to previous